doc/development/adding_templates.rst

```diff
@@ -0,0 +1,99 @@
+:mod:`state_preparations <pennylane.templates.state_preparations>` or
+:mod:`soubroutines <pennylane.templates.soubroutines>`.
+Here we will use the template ``MyNewTemplate`` as an example, and you need to replace ``<templ_type>`` with the
+correct template type.
+
+1. **Add the template** by adding a new file ``my_new_template.py`` to the correct ``templates/<templ_type>/``
+   subdirectory, which contains your new template. For example, this is a very basic template applying an ``qml.RX``
+   gate to each wire:
+
+   .. code-block:: python
+
+        import pennylane as qml
+        from pennylane.templates import template  # import the decorator
+
+        @template
+        def MyNewTemplate(weights, wires):
+
+            # Check that the inputs have the correct format
+            # ...
+
+            for wire, weight in zip(wires, weights):
+                qml.RX(weight, wires=wire)
+
+   A template is a function that defines a sequence of quantum gates (without measurements).
+   Since the template is called within a :ref:`quantum function <intro_vcirc_qfunc>`,
+   it can only contain information processing that is allowed
+   inside a quantum functions.
+
+   Make sure you consider the following:
+
+   * To be consistent with other quantum operations (which are classes), the function name (i.e., ``MyNewTemplate``) is
+     written in camel case. Choose the name carefully. Good names tell the user what a template is good for, or what architecture
+     it implements.
+
+   * A template is "registered" by using the :mod:`template <pennylane.template>` decorator ``@template``.
+     This allows us to record the queue of operations of a template, which is very useful for testing:
+
+     .. code-block:: python
+
+        with qml.utils.OperationRecorder() as rec:
+            MyNewTemplate(...)
+
+        list_of_gates = rec.queue
+
+   * Consider using the :func:`broadcasting <pennylane.templates.broadcasting>` function to make your code more concise.
+
+   * Write an extensive docstring that explains how to use the template. Include a sketch of the template (add the
+     file to the ``doc/_static/templates/<templ_type>/`` directory). You can also display a small usage example
+     at the beginning of the docstring.
+
+     At the end of the docstring, add a section starting with the ``.. UsageDetails::`` directive,
+     where you demonstrate with code examples how to use the templates with different
+     settings, for example varying the number of wires, explaining keyword arguments and special cases.
+     For inspiration, check one of the existing templates, such as
+     :func:`AmplitudeEmbedding <pennylane.templates.embeddings.AmplitudeEmbedding>`
+
+   * Check the inputs to the template. You can use the functions provided in :mod:`utils <pennylane.templates.utils>`.
+     Don't forget that arguments may be passed by the user to the qnode as positional or keyword arguments, and
```

Comment:
I'm assuming that this was originally meant to refer to differentiability of arguments in PL. Something like the suggestion (though that might be the very first time a user sees these terms) or explicitly stating the difference beforehand could work well I guess.
```suggestion
     Don't forget that arguments may be passed by the user to the qnode as primary or auxiliary quantum function arguments, and
```